packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -73,7 +73,7 @@ export class Vector2 {
      * @param index defines the offset in source array
      * @returns the current Vector2
      */
-    public toArray(array: FloatArray, index: number = 0): Vector2 {
+    public toArray(array: FloatArray, index: number = 0): this {

```

Comment:
Not an expert but I presume that this is not recognized as a type. You are still returning a Vector2 so it should remain as a Vector2 even though you are returning it in the form of this (a Vector2)

Same where you have changed other types to this rather than a named type.